tools/redpanda-gdb.py

```diff
@@ -90,10 +111,24 @@ def __init__(self, ref):
         self.element_size_bytes = self.element_type.sizeof
 
         end_cap_type = self.ref["__end_cap_"].type.strip_typedefs()
+
         end_cap_type = end_cap_type.template_argument(0)
         end_cap_type_fmt = "std::__1::__compressed_pair_elem<{}, 0, false>"
-        self.end_cap_type = gdb.lookup_type(
-            end_cap_type_fmt.format(end_cap_type))
+        try:
+            self.end_cap_type = gdb.lookup_type(
+                end_cap_type_fmt.format(end_cap_type))
+        except:
+            # Try converting "struct foo *" into "foo*": sometimes GDB reports the type
```

Comment:
is this because c++ let's us do `struct foo {};` then `struct foo*` or `foo*` but gdb doesn't normalize it? either way, nice catch.

Reply:
I'm not sure if it's a GDB bug or something weird in the debug symbols, but the type that comes out of `template_argument(0)` is like "struct foo *", and the name in `info types` is like "foo*".  Mysteries.